var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -80,32 +80,56 @@ def install(self, spec, prefix):
             optflags = {
                 'gcc': ['-O2',
                         '-ffast-math',
-                        '-ffree-form',
-                        '-ffree-line-length-none',
                         '-ftree-vectorize',
                         '-funroll-loops',
                         '-mtune=native'],
                 'intel': ['-O2',
                           '-pc64',
-                          '-unroll',
-                          '-heap-arrays 64']
+                          '-unroll']
             }
+
+            dflags = ['-DNDEBUG']
+
             cppflags = [
                 '-D__FFTW3',
                 '-D__LIBINT',
                 spec['fftw'].headers.cpp_flags
             ]
+
+            if '^mpi@3:' in spec:
+                cppflags.append('-D__MPI_VERSION=3')
+            elif '^mpi@2:' in spec:
+                cppflags.append('-D__MPI_VERSION=2')
+
+            if '^intel-mkl' in spec:
+                cppflags.append('-D__FFTSG')
+
+            cflags = copy.deepcopy(optflags[self.spec.compiler.name])
+            cxxflags = copy.deepcopy(optflags[self.spec.compiler.name])
             fcflags = copy.deepcopy(optflags[self.spec.compiler.name])
-            fcflags.append(spec['fftw'].headers.cpp_flags)
+            fcflags.extend([
+                '-ffree-form',
+                '-ffree-line-length-none',
+                spec['fftw'].headers.cpp_flags
+            ])
+
+            if '%intel' in spec:
+                cflags.append('-fp-model precise')
+                cxxflags.append('-fp-model precise')
+                fcflags.extend(['-fp-model source', '-heap-arrays 64'])
```

Comment:
To extend on this: `cp2k` builds using a home-baked Makefile system,

They expect to source a sort of `make.inc` based on "ARCH" and "VERSION" parameters passed on the command line. These two parameters translate in the name of the file to be sourced under the directory `arch`. Some of those files just set a bunch of variables and include other files with the logic. Finally they expect you to tweak manually the location of certain libraries, or what goes in certain flags.

Not knowing the source code, I tried to get it working by writing a similar file to be included, reusing everything I could from the flags in the original files.